File: src/main/webapp/app/model/SubInfo.js
Ext.define(App.path('model.SubInfo'), {
	extend : 'Ext.data.Model',
	config : {
		fields : [ {
			name : 'fullname'
		}, {
			name : 'msisdn'
		}, {
			name : 'subid'
		}, {
			name : 'subid_provided_place'
		}, {
			name : 'subid_address'
		}, {
			name : 'id_front_img'
		}, {
			name : 'id_back_img'
		}, {
			name : 'sim_img'
		}, {
			name : 'contact'
		}, {
			name : 'serialsim'
		}, {
			name : 'simcode'
		}, ],
		validations : [ {
			type : 'presence',
			field : 'fullname',
			message : "Họ tên chủ TB chưa được nhập!"
		}, {
			type : 'presence',
			field : 'subid',
			message : "Số CMT chưa được nhập!"
		}, {
			type : 'presence',
			field : 'contact',
			message : "Điện thoại/Email liên hệ chưa được nhập!"
		}, {
			type : 'presence',
			field : 'id_front_img',
			message : "Ảnh CMT mặt trước chưa được chọn!"
		}, {
			type : 'presence',
			field : 'id_back_img',
			message : "Ảnh CMT mặt sau chưa được chọn!"
		}, {
			type : 'presence',
			field : 'sim_img',
			message : "Ảnh phôi sim chưa được chọn!"
		}, {
			type : 'presence',
			field : 'subid_address',
			message : "Địa chỉ theo CMT chưa được nhập!"
		}, {
			type : 'presence',
			field : 'subid_provided_place',
			message : "Nơi cấp CMT chưa được nhập!"
		}, {
			type : 'presence',
			field : 'msisdn',
			message : "Số điện thoại đăng ký chưa được nhập!"
		}, {
			type : 'presence',
			field : 'serialsim',
			message : "Serial SIM chưa được nhập!"
		}, {
			type : 'presence',
			field : 'simcode',
			message : "Mã bảo mật chưa được nhập!"
		}, {
			type : 'format',
			field : 'msisdn',
			matcher : /099\d{7}|0199\d{7}|8499\d{7}|84199\d{7}/,
			message : "Không đúng định dạng số điện thoại Gmobile."
		}, 
//		{
//			type : 'format',
//			field : 'subid',
//			matcher : /\d{10}|\d{11}|\d{12}|\d{13}/,
//			message : "Số CMT không đúng định dạng là số."
//		}, 
			
		]
	}

});

File: src/main/java/vn/gmobile/subinfo_reg/model/SubInfo.java
package vn.gmobile.subinfo_reg.model;



import java.util.Date;

import com.bean.annot.*;
import com.bean.base.Bean;
import com.bean.base.BeanProperty;



@Entity(name = "subinfo")
public class SubInfo extends Bean {
	@Attribute(name = "id")
	@ExternalKey
//	@AutoIncrement
	@Final
	public static final BeanProperty<Integer> id =  BeanProperty.integerType();
	@Attribute(name = "subid")
	public static final BeanProperty<String> subid =  BeanProperty.stringType();
	@Attribute(name = "fullname")
	public static final BeanProperty<String> fullname =  BeanProperty.stringType();
	@Attribute(name = "msisdn")
	public static final BeanProperty<String> msisdn =  BeanProperty.stringType();
	@Attribute(name = "sex")
	public static final BeanProperty<Integer> sex =  BeanProperty.integerType();
	@Attribute(name = "dob")
	public static final BeanProperty<Date> dob =  BeanProperty.dateType();
	@Attribute(name = "subid_provided_date")
	public static final BeanProperty<Date> subid_provided_date =  BeanProperty.dateType();
	@Attribute(name = "subid_provided_place")
	public static final BeanProperty<String> subid_provided_place =  BeanProperty.stringType();
	@Attribute(name = "subid_address")
	public static final BeanProperty<String> subid_address =  BeanProperty.stringType();
	@Attribute(name = "friend_msisdn")
	public static final BeanProperty<String> friend_msisdn =  BeanProperty.stringType();
	@Attribute(name = "id_front_img")
	public static final BeanProperty<String> id_front_img =  BeanProperty.stringType();
	@Attribute(name = "id_back_img")
	public static final BeanProperty<String> id_back_img =  BeanProperty.stringType();
	@Attribute(name = "sim_img")
	public static final BeanProperty<String> sim_img =  BeanProperty.stringType();
	
	@Attribute(name = "contact")
	public static final BeanProperty<String> contact =  BeanProperty.stringType();
	@Attribute(name = "active_date")
	public static final BeanProperty<Date> active_date =  BeanProperty.dateType();
	@Attribute(name = "recent_charge_date")
	public static final BeanProperty<Date> recent_charge_date =  BeanProperty.dateType();
	@Attribute(name = "recent_charge_value")
	public static final BeanProperty<Integer> recent_charge_value =  BeanProperty.integerType();
	
	@Attribute(name = "otp_in_day")
	public static final BeanProperty<Integer> otp_in_day =  BeanProperty.integerType();
	@Attribute(name = "info_update_date")
	public static final BeanProperty<Date> info_update_date =  BeanProperty.dateType();
	@Attribute(name = "expr_date")
	public static final BeanProperty<Date> expr_date =  BeanProperty.dateType();
	@Attribute(name = "login_fail")
	public static final BeanProperty<Integer> login_fail =  BeanProperty.integerType();
	@Attribute(name = "otp")
	public static final BeanProperty<String> otp =  BeanProperty.stringType();
	
	@Attribute(name = "serialsim")
	public static final BeanProperty<String> serialsim =  BeanProperty.stringType();
	@Attribute(name = "simcode")
	public static final BeanProperty<String> simcode =  BeanProperty.stringType();
	@Attribute(name = "requesttype")
	public static final BeanProperty<Integer> requesttype =  BeanProperty.integerType();
	
	@Attribute(name = "status")
	public static final BeanProperty<Integer> status =  BeanProperty.integerType();
	@Attribute(name = "from_address")
	public static final BeanProperty<String> from_address =  BeanProperty.stringType();
}
